packages/workbox-precaching/controllers/PrecacheController.mjs

```diff
@@ -190,6 +194,40 @@ class PrecacheController {
     };
   }
 
+  /**
+   * Takes the current set of temporary files and moves them to the final
+   * cache, deleting the temporary cache once copying is complete.
+   *
+   * @return {
+   * Promise<workbox.precaching.CleanupResult>}
+   * Resolves with an object containing details of the deleted cache requests
+   * and precache revision details.
+   */
+  async activate() {
+    const tempCache = await caches.open(this._getTempCacheName());
+
+    const requests = await tempCache.keys();
+    await Promise.all(requests.map(async (request) => {
+      const response = await tempCache.match(request);
+      await cacheWrapper.put(this._cacheName, request, response);
```

Comment:
We have https://github.com/GoogleChrome/workbox/issues/1312 open to track the larger question of how to deal with quota issues that come up during precaching.

In the meantime, this PR introduces some overhead due to every response in the temp cache being duplicated first, and with all the temp entries only getting deleted after all of those duplicates are written.

What if you added in

```js
await cacheWrapper.delete(this._getTempCacheName(), request);`
```

inside this loop, to clear out the temporary copy of each individual response after it's been duplicated to the final array?

You can still leave the

```js
await caches.delete(this._getTempCacheName());
```

outside the loop, to delete the cache itself, though at that point the cache should have 0 entries in it.